src/content/contribute/writing-a-scaffold.md

```diff
@@ -0,0 +1,394 @@
+      Input('entry', 'What is the entry point in your app?')
+    ]).then(answer => {
+      if (answer['confirm'] === 'Pengwings') {
+        this.options.env.configuration.dev.webpackOptions = createDevConfig(answer);
+      }
+    });
+  }
+};
+```
+
+Let's go ahead and create our config.
+
+
+## Some more configs
+
+Let's start by looking at `dev-config.js`. We have some answers, now we want to use them to build up a config. We will start with mounting our values on the variables we've declared, with some properties we want to build up. We also want to use the answers for the `entry` property. We've also added an output property that has a `filename`.
+
+T> String values must be quoted twice. This is to preserve our ability to add other functionality, using only " ", while " 'Mystring' " resolves to a string.
+
+__dev-config.js__
+
+```js
+module.exports = function createDevConfig(answer) {
+  let entryProp = answer.entry ? ( '\'' + answer.entry + '\'') : '\'index.js\'';
+  let devConfig = {
+    entry: entryProp,
+    output: {
+      filename: '\'[name].js\''
+    }
+  };
+  return devConfig;
+};
+
+```
+
+Run `webpack init webpack-scaffold-demo`, and you should see scaffold working.
+
+
+## Basic Scaffold
+
+Now that we've got our initial scaffold. Let's add the rest of our options! For the `context`, let's say we've got a `path.join` we want to make use of. For this, we use a single quote string. By default the current directory is used, but it's recommended to pass a value in your configuration (context). This makes your configuration independent from CWD (current working directory).
+
+```js
+module.exports = function createDevConfig(answer) {
+  let entryProp = answer.entry ? ( '\'' + answer.entry + '\'') : '\'index.js\'';
+  let devConfig = {
+    entry: entryProp,
+    output: {
+      filename: '\'[name].js\''
+    },
+    context: 'path.join(__dirname, "src")'
+  };
+  return devConfig;
+};
+```
+
+
+## Add more functionality
+
+Now we are ready to add a plugin. For this, let's create an utility for [`SplitChunksPlugin`](/plugins/split-chunks-plugin/) based on the input from the user. Start by adding another question to our prompt.
+
+```js
+const Generator = require('yeoman-generator');
+const List = require('@webpack-cli/webpack-scaffold').List;
+const Input = require('@webpack-cli/webpack-scaffold').Input;
+const createDevConfig = require('./dev-config');
+
+module.exports = class WebpackGenerator extends Generator {
+  constructor(args, opts) {
+    super(args, opts);
+    opts.env.configuration = {
+      dev: {
+        webpackOptions: {}
+      }
+    };
+  }
+
+  prompting() {
+    return this.prompt([
+      List('confirm', 'Welcome to the demo scaffold! Are you ready?', ['Yes', 'No', 'Pengwings']),
+      Input('entry', 'What is the entry point in your app?'),
+      Input('plugin', 'What do you want to name your splitChunks?')
```

Comment:
Maybe pick some other plugin that is easier to use as an example? splitting your code can be done by using optimization.splitChunks now without specifically providing a plugin to the config.

Reply:
Sounds great, will replace the splitChunks with html-webpack-plugin